ddtrace/internal/writer/writer.py

```diff
@@ -420,14 +421,17 @@ def _flush_queue_with_client(self, client, raise_exc=False):
                 )
         finally:
             if config.health_metrics_enabled and self.dogstatsd:
+                # _metrics is updated when traces are queued.
+                # Here we get a deep copy of the nested metrics dictionary as early as possible
+                metrics = copy.deepcopy(self._metrics)
```

Comment:
If after `deepcopy` there is change to `self._metircs`, those change will lose due to `self._metrics_reset()` in `self.flush_queue`?

Reply:
Yeah, maybe it'd be better to `deepcopy` in `flush_queue`, immediately call `_metrics_reset()`, then pass the copied object to `_flush_queue_with_client()`.

That said, metrics are somewhat best-effort (eg: `self._metrics_reset()` is called in the `finally:` above, so we clear metrics no matter what).

Reply:
LGTM.